src/modules/routing/repository/MediationRecord.ts

```diff
@@ -0,0 +1,93 @@
+import type { TagsBase } from '../../../storage/BaseRecord'
+import type { MediationRole } from '../models/MediationRole'
+import type { Verkey } from 'indy-sdk'
+
+import { AriesFrameworkError } from '../../../error'
+import { BaseRecord } from '../../../storage/BaseRecord'
+import { uuid } from '../../../utils/uuid'
+import { MediationState } from '../models/MediationState'
+
+export interface MediationRecordProps {
+  id?: string
+  state: MediationState
+  role: MediationRole
+  createdAt?: Date
+  connectionId: string
+  endpoint?: string
+  recipientKeys?: Verkey[]
+  routingKeys?: Verkey[]
+  default?: boolean
+  tags?: CustomMediationTags
+}
+
+export type CustomMediationTags = TagsBase
+export type DefaultMediationTags = {
+  role: MediationRole
+  connectionId: string
+  state: MediationState
+}
+
+export class MediationRecord
+  extends BaseRecord<DefaultMediationTags, CustomMediationTags>
+  implements MediationRecordProps
+{
+  public state!: MediationState
+  public role!: MediationRole
+  public connectionId!: string
+  public endpoint?: string
+  public recipientKeys!: Verkey[]
+  public routingKeys!: Verkey[]
+  public default = false
```

Comment:
I think this could be improved. I like being able to recover the default mediator from a reboot so I don't think the config is the best place for it. I played with having default as a tag on the record but decided to move it inside of the record during a debug session. I am happy to go in a different direction if desired.